components/tidb_query/src/rpn_expr/types/expr_builder.rs

```diff
@@ -307,6 +309,7 @@ where
 
     let args: Vec<_> = tree_node.take_children().into();
     let args_len = args.len();
+    let data = (func_meta.data_initializer_ptr)(&tree_node);
```

Comment:
I think it's better to move the data initialization above L335 and make `RpnExpressionNode::FnCall` as the parameter of `func_meta.data_initializer_ptr` to allow `func_meta.data_initializer_ptr` use `RpnExpressionNode`'s which on the stack.